packages/guilded.js/lib/structures/Message.ts

```diff
@@ -177,6 +173,29 @@ export class Message extends Base<ChatMessagePayload> {
   send(content: MessageContent) {
     return this.client.messages.send(this.channelId, content);
   }
+  /** Returns the url of this message */
+  async url(): Promise<string> {
+    const channel: Channel | null = await this.channel();
+    const server = await this.server();
+    // Will Update once we have a way to get the DM env
+    if (!server) return "";
+    return `https://www.guilded.gg/${server.name}/groups/${channel.groupId}/channels/${channel.id}/chat?messageId=${this.id}`;
+  }
+  /**
+   * Returns the channel that this message belongs to
+   */
+  channel(): Channel | Promise<Channel> {
+    return (
+      this.client.channels.cache.get(this.channelId) ??
+      this.client.channels.fetch(this.channelId)
+    );
+  }
+  server(): Server | Promise<Server | null> {
+    return (
+      this.client.servers.cache.get(this.serverId || "") ??
+      this.client.servers.fetch(this.serverId || "").catch(() => null)
+    );
+  }
```

Comment:
These methods do not follow our type standards for the package. Return types should not be complex unions like a Structure | Promise<Structure>. We also don't want to take any control flow from the user who decides when they should fetch or not. Therefore, this method will be restored to the cache reliant version.